src/git.rs

```diff
@@ -22,15 +27,41 @@ pub fn repo_hash<P: AsRef<Path>>(path: P) -> Option<String> {
     }
 }
 
-fn dirty<P: AsRef<Path>>(path: P) -> bool {
-    let path = path.as_ref().to_str();
-    let mut args = match path {
-        Some(path) => vec!["-C", path],
-        None => vec![],
-    };
-    args.extend(&["ls-files", "-m"]);
-    match Command::new("git").args(&args).output() {
+fn dirty(path: impl AsRef<Path>) -> bool {
+    match git_command(path.as_ref(), &["ls-files", "-m"]) {
         Ok(modified_files) => !modified_files.stdout.is_empty(),
         Err(_) => false,
     }
 }
+
+fn gitmodules_config(subcommand: &[&str]) -> Option<String> {
+    let mut args = vec!["config", "-f", ".gitmodules", "-z"];
+    args.extend(subcommand);
+    let output = git_command(Path::new("."), &args).ok()?;
+
+    if !output.status.success() {
+        return None;
+    }
+
+    let mut result = String::from_utf8(output.stdout).ok()?;
+    assert_eq!(result.pop(), Some('\0'));
```

Comment:
I swear there should be something better for this _while retaining an **owned** string_ (`strip_suffix` and friends don't work).

Reply:
Where does the `\0` come from and why didn't that cause problems for any of the other git commands before?

Reply:
`-z` makes it so that the string is terminated by `\0`. Now that no string splitting is needed anymore it's probably fine to leave it terminated by `\n`. The existing `repo_hash` uses `.trim_end_matches('\n')` which returns a slice reference and has to be cloned again, when we really want to resize the underlying `Vec<u8>` of a string (size -= 1, capacity stays the same).

Oh well, performance. Might as well use that and `.into()`/`.to_string()`/`.to_owned()`.

Reply:
Makes sense